src/main/java/org/jabref/gui/actions/CleanupAction.java

```diff
@@ -39,6 +43,26 @@ public CleanupAction(BasePanel panel, JabRefPreferences preferences) {
     }
 
     @Override
+    public void action() {
+        init();
+        Platform.runLater(() -> {
+            if (canceled) {
+                return;
+            }
+            CleanupDialog cleanupDialog = new CleanupDialog(panel.getBibDatabaseContext(), preferences.getCleanupPreset());
+
+            Optional<CleanupPreset> chosenPreset = cleanupDialog.showAndWait();
+            chosenPreset.ifPresent(cleanupPreset ->
+                    BackgroundTask
+                            .wrap(() -> {
+                                cleanup(cleanupPreset);
+                                return null;
```

Comment:
Please copy the changes from https://github.com/JabRef/jabref/pull/4119/files#diff-851966b1f29f8a595c4b262cf1f895bb (it may take a while until this PR is merged). Then these `return null` statements are no longer needed (in several places).